### PR TITLE
Interim merge for #1475

### PR DESCRIPTION
**Pull request by randomeizer**
@latenitefilms This is a stable build on the pathway to #1475 (for me). Can you test when you get a chance, and we'll merge with develop. I still have work to do on the issue, so don't delete the branch.

And if you get time, run `_test("cp.apple.finalcutpro")()` overnight (it will take about an hour to complete I think) and log the results?

### CODE REVIEW

File: src/extensions/middleclass/init.lua [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      end
+      subclass.initialize = function(instance, ...) return self.initialize(instance, ...) end
+
+      self.subclasses[subclass] = true
+      self:subclassed(subclass)
+
+      return subclass
+    end,
+
+    subclassed = function(self, other) end,
```
[the remark this answers is not in the document]
Comment (randomeizer) [reply src/extensions/middleclass/init.lua:160]: External library.

[the remark this answers is not in the document]
Comment (latenitefilms) [reply src/extensions/middleclass/init.lua:160]: Add a `-- luacheck: ignore`?

[the remark this answers is not in the document]
Comment (randomeizer) [reply src/extensions/middleclass/init.lua:160]: I'd prefer to leave it unmodified I think. Can just ignore stickler in this merge.

**Comment by latenitefilms** [comment]
@randomeizer - Test results:

```
CommandPost v1.0.0-beta.33

Date Built:                     Sep  4 2018, 14:10:31
macOS Version:                  10.13.6
CommandPost Locale:             en
Current Keyboard Layout:        Australian
Final Cut Pro Path:             /Applications/Final Cut Pro.app
Final Cut Pro Version:          10.4.3
Final Cut Pro Language:         en
Developer Mode:                 true

2018-09-04 15:05:33: 2018-09-04 15:05:32: -- Loading /Users/latenitealpha/CommandPost/src/extensions/cp/init.lua
2018-09-04 15:05:32: -- Loading extension: uielement
2018-09-04 15:05:33: 15:05:33         cp:     Loading Plugins...
2018-09-04 15:05:35: 15:05:35                 Plugins Loaded.
2018-09-04 15:05:35: -- Done.

> _test("cp.apple.finalcutpro")()
2018-09-04 15:05:52: 
2018-09-04 15:05:52: [ERROR] cp.apple.finalcutpro > zh_CN > Launch FCP: Error occurred before test 'Launch FCP': ...pha/CommandPost/src/tests/cp/apple/finalcutpro/_test.lua:639: Unable to open the 'Test - Launch FCP' Library.
stack traceback:
	[C]: in function 'error'
	...pha/CommandPost/src/tests/cp/apple/finalcutpro/_test.lua:639: in function 'cp.apple.finalcutpro._test._beforeEach'
	(...tail calls...)
	[C]: in function 'xpcall'
	...atenitealpha/CommandPost/src/extensions/cp/test/init.lua:426: in local 'runTests'
	...pha/CommandPost/src/tests/cp/apple/finalcutpro/_test.lua:576: in function 'cp.apple.finalcutpro._test._run'
	...atenitealpha/CommandPost/src/extensions/cp/test/init.lua:420: in function <...atenitealpha/CommandPost/src/extensions/cp/test/init.lua:413>
	(...tail calls...)
	[C]: in function 'xpcall'
	...ealpha/CommandPost/src/extensions/hs/_coresetup/init.lua:524: in function <...ealpha/CommandPost/src/extensions/hs/_coresetup/init.lua:500>
2018-09-04 15:15:22: 15:15:22 ** Warning:      menu: Unable to match step #3 in { "Window", "Show in Workspace", "Event Viewer" }, a string with a value of "Event Viewer" (en) with the app in cp.i18n.localeID: German [de]
2018-09-04 15:15:22: 
2018-09-04 15:15:22:  [FAIL] cp.apple.finalcutpro > de > Event Viewer: [...pha/CommandPost/src/tests/cp/apple/finalcutpro/_test.lua:83] 
2018-09-04 15:16:54: 15:16:54 ** Warning:      menu: Unable to match step #2 in { "View", "Toggle Inspector Height" }, a string with a value of "Toggle Inspector Height" (en) with the app in cp.i18n.localeID: German [de]
2018-09-04 15:23:57: 
2018-09-04 15:23:57: [RESULT] cp.apple.finalcutpro: passed: 1462; failed: 1; errors: 1; time: 30.8431s

2018-09-04 15:23:58: ********
2018-09-04 15:23:58: 15:23:58 ERROR:   LuaSkin: hs.timer callback error: /Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: Timed out after 20000 ms.
stack traceback:
	[C]: in function 'error'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: in method 'onError'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3028: in function </Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3003>
2018-09-04 15:23:58: ********
2018-09-04 15:23:58: ********
2018-09-04 15:23:58: 15:23:58 ERROR:   LuaSkin: hs.timer callback error: /Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: Timed out after 20000 ms.
stack traceback:
	[C]: in function 'error'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: in method 'onError'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3028: in function </Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3003>
2018-09-04 15:23:58: ********
2018-09-04 15:23:58: ********
2018-09-04 15:23:58: 15:23:58 ERROR:   LuaSkin: hs.timer callback error: /Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: Timed out after 20000 ms.
stack traceback:
	[C]: in function 'error'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: in method 'onError'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3028: in function </Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3003>
2018-09-04 15:23:58: ********
2018-09-04 15:23:58: ********
2018-09-04 15:23:58: 15:23:58 ERROR:   LuaSkin: hs.timer callback error: /Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: Timed out after 20000 ms.
stack traceback:
	[C]: in function 'error'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: in method 'onError'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3028: in function </Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3003>
2018-09-04 15:23:58: ********
2018-09-04 15:23:58: ********
2018-09-04 15:23:58: 15:23:58 ERROR:   LuaSkin: hs.timer callback error: /Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: Timed out after 20000 ms.
stack traceback:
	[C]: in function 'error'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: in method 'onError'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3028: in function </Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3003>
2018-09-04 15:23:58: ********
2018-09-04 15:24:17: ********
2018-09-04 15:24:17: 15:24:17 ERROR:   LuaSkin: hs.timer callback error: /Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: Timed out after 20000 ms.
stack traceback:
	[C]: in function 'error'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:202: in method 'onError'
	/Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3028: in function </Users/latenitealpha/CommandPost/src/extensions/cp/rx.lua:3003>
2018-09-04 15:24:17: ********
```

**Pull request merged**